### DISCUSSION TITLE
Adding easybuild support

### DISCUSSION THREAD

**Discussion by djarecka** [ReproNim/neurodocker#546] [Ideas]
During the meeting we had discussion about the [EESSI project](http://www.eessi.io/docs/software_testing/#setting-up-a-reframe-configuration-file) and how we can use it to build software from source. Below I copy notes from our meetings as a starting point, but would be good to continue the discussion:

- Focussed on performance, everything compiled from source optimizing to the architecture
- Can generate docker/singularity recipe (replacement for compiled from source?)
- Example for fsl: https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs/f/FSL 
- Example for afni: https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs/a/AFNI 
- Works on all linux distribution
- Has matlab recipe, e.g. https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/SPM/SPM-12.5_r7771-MATLAB-2021b.eb
- Neurodocker would hide the eessi complexity 
- EESSI project uses easybuild to build a distribution of software for HPCs and just got funded in the EU
- Adding option to build with easybuild and test

**Comment by kaczmarj** [comment]
i don't have the bandwidth to work on this now unfortunately, but i'm very excited about the idea of incorporating easybuild to build neuroimaging packages from source. it would likely solve a longstanding issue in neurodocker --- building images with multiple neuroimaging packages that may have conflicting dependencies.

perhaps one could create a minimal example of how to build, let's say, a container with ANTs built from source in a recent Debian or Ubuntu image.
